huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go

```diff
@@ -405,49 +412,50 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
-	config := meta.(*config.Config)
-	client, err := config.BcsV2Client(GetRegion(d, config))
+func resourceBCSInstanceV2Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.BcsV2Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud Blockchain client: %s", err)
+		return diag.Errorf("error creating blockchain client: %s", err)
 	}
 
 	instanceID := d.Id()
 	instance, err := blockchains.Get(client, instanceID).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error getting Blockchain instance (%s): %s", instanceID, err)
-	}
-	logp.Printf("[DEBUG] Retrieved Blockchain instance %s: %#v", instanceID, instance)
-
-	d.Set("name", instance.Basic.Name)
-	d.Set("edition", instance.Basic.VersionType)
-	d.Set("blockchain_type", instance.Basic.ServiceType)
-	d.Set("consensus", instance.Basic.Consensus)
-	d.Set("security_mechanism", instance.Basic.SignAlgorithm)
-	d.Set("database_type", instance.Basic.DatabaseType)
-	d.Set("restful_api_support", instance.Basic.IsSupportRestful)
-	d.Set("status", instance.Basic.Status)
-	d.Set("tc3_need", instance.CouchDB != (blockchains.CouchDB{}))
-	d.Set("cluster_type", instance.Basic.ClusterType)
-	d.Set("cce_cluster_id", instance.Basic.ClusterID)
-	d.Set("version", instance.Basic.Version)
-	d.Set("purchase_type", instance.Basic.PurchaseType)
-	d.Set("cross_region_support", instance.Basic.IsCrossRegion)
-	d.Set("rollback_support", instance.Basic.IsSupportRollback)
-	d.Set("old_service_version", instance.Basic.OldServiceVersion)
+		return common.CheckDeletedDiag(d, err, "error getting blockchain instance")
+	}
+	log.Printf("[DEBUG] retrieved blockchain instance %s: %#v", instanceID, instance)
+
+	mErr := multierror.Append(
+		d.Set("name", instance.Basic.Name),
+		d.Set("edition", instance.Basic.VersionType),
+		d.Set("blockchain_type", instance.Basic.ServiceType),
+		d.Set("consensus", instance.Basic.Consensus),
+		d.Set("security_mechanism", instance.Basic.SignAlgorithm),
+		d.Set("database_type", instance.Basic.DatabaseType),
+		d.Set("restful_api_support", instance.Basic.IsSupportRestful),
+		d.Set("status", instance.Basic.Status),
+		d.Set("tc3_need", instance.CouchDB != (blockchains.CouchDB{})),
+		d.Set("cluster_type", instance.Basic.ClusterType),
+		d.Set("cce_cluster_id", instance.Basic.ClusterID),
+		d.Set("version", instance.Basic.Version),
+		d.Set("purchase_type", instance.Basic.PurchaseType),
+		d.Set("cross_region_support", instance.Basic.IsCrossRegion),
+		d.Set("rollback_support", instance.Basic.IsSupportRollback),
+		d.Set("old_service_version", instance.Basic.OldServiceVersion),
+	)
 
 	portalAddrs := make([]string, len(instance.Basic.AgentPortalAddress))
-	for i, v := range instance.Basic.AgentPortalAddress {
-		portalAddrs[i] = v
-	}
+	copy(portalAddrs, instance.Basic.AgentPortalAddress)
```

Comment:
we can save instance.Basic.AgentPortalAddress directly
array of simple types, like []string and []int can be used in d.Set directly

Reply:
ok, but lint suggest should use copy(to, from) instead of a loop

Reply:
done